jms/1.7/modules/ROOT/pages/jms-publish-consume.adoc

```diff
@@ -1,82 +1,106 @@
-`replyToDestination` queue until a message is consumed or the `maximumWait` is
-reached, in which case it fails with a `JMS:TIMEOUT` error.
 
-== Request Reply Patterns
 
-As of JMS Connector 1.6.0:
+== Configure Request and Reply Patterns
+
+In JMS, exists two well-known request and reply patterns to understand how request and reply messages are correlated. In the *Request-Reply Pattern* field of the operation you can configure the *CORRELATION_ID (Default)* and *MESSAGE_ID* patterns. Both ensure that an incoming message is the proper reply of an outgoing message.
+You can also configure the field to *NONE* to not use any of the patterns, and the consumer tries to consume from the reply queue without using any selector to correlate messages. This option is useful when working with temporal queues that ensure that the queue has only one message.
+
+=== Configure the Correlation ID Pattern
 
-In JMS, there exists two well know Request-Reply patterns for how request and reply messages
-are correlated. This ensures that an incoming message is the proper reply
-of an outgoing message.
+The *CORRELATION_ID (Default)* pattern implies that the operation after publishing a message to the destination queue, consumes a message using a selector that expects a message with the same correlation id to be used to publish the request message.
 
-=== Correlation ID Pattern
+In the following example, in the first flow you configure the *Publish consume* operation to publish a message to the `targetDestination` with the current `payload` as its body and the JMSReplyTo header configured to the known queue setting the *Reply To* field to `replyToDestination`. Also, the operation has configured the *Request-Reply Pattern* field to *CORRELATION_ID (Default)*. +
+Then, in the second flow an *On New Message* source reads the published message and responds using the same correlation id pattern. A *Logger* component, logs the message response. +
+Back in the first flow, the *Publish consume* operation consumes the response received and then another *Logger* component, logs the final message.
 
-This pattern is the default one in the *Publish Consume* operation, the usage of
-this pattern implies that the operation after publishing a message to the
-destination queue, consumes a message using a selector that expects a message
-with the same Correlation ID to be used to publish the request message.
+. In your Studio flow, select the *Publish consume* operation.
+. In the connector configuration screen, set the *Destination* field to `targetDestination`.
+. Set the *Request-Reply Pattern* to *CORRELATION_ID (Default)*.
+. In the *Message* section, set the *Reply To* field to *Edit inline*.
+. Set the *Destination Name* field to `replyToDestination`.
+. Drag a *Logger* component to right of the Publish consume.
+. Set the *Message* field to `Received message from 'replyToDestination' with Correlation ID: #[attributes.headers.correlationId]`.
```

Comment:
The way this is tagged makes it look like the message is:

`Received message from 'replyToDestination' with the Correlation ID: #[attributes.headers.correlationId]`

Is that right?